docs/GUIDE.md

```diff
@@ -0,0 +1,123 @@
+```sbt
+libraryDependencies += "org.virtuslab.ash" %% "circe-akka-serializer" % "Version"
+```
+
+## How to use:
+1. [Circe Akka Serializer](#circe-akka-serializer-guide)
+2. [Annotations](#annotations-guide)
+3. [Serializability Checker Compiler Plugin](#serializability-checker-compiler-plugin-guide)
+4. [Codec Registration Checker Compiler Plugin](#codec-registration-checker-compiler-plugin-guide)
+5. [Dump Persistence Schema Compiler Plugin](#dump-persistence-schema-compiler-plugin-guide)
+6. [ashDumpPersistenceSchema sbt task](#ashdumppersistenceschema-sbt-task)
+
+### Circe Akka Serializer Guide
+[CirceAkkaSerializer](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala) is the main abstract class that you have to extend to use as the Serializer in your application. Before extending it, read Javadoc comments available in following files:
+- [CirceAkkaSerializer.scala](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala) (the most important starting point / main abstraction)
+- [CirceTraitCodec.scala](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceTraitCodec.scala) (a trait that is extended by CirceAkkaSerializer and has some important `val`s that should be overridden in your code)
+- [Register.scala](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/Register.scala) (helper object with `apply` method that needs to be used in order to register Codecs for serializable type)
+- [Registration.scala](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/Registration.scala) (`Register.apply` method returns an instance of type Registration. Registration instances are later used as registered codecs)
+
+Javadocs from these files and a quick lecture of [Akka serialization guide](https://doc.akka.io/docs/akka/current/serialization.html) should be enough to use Circe Akka Serializer properly. Apart from implementing the Circe Akka Serializer in scala code, remember to configure your Serializer in Akka-specific configuration file (.conf file). Two mandatory configurations are `akka.actor.serializers` and `akka.actor.serialization-bindings` (example below):
+```editorconfig
+akka {
+  actor {
+    serializers {
+      circe-json = "org.example.ExampleSerializer"
+    }
+    serialization-bindings {
+      "org.example.MySerializable" = circe-json
+    }
+  }
+}
+```
+where "org.example.ExampleSerializer" is the FQCN of your Serializer which extends CirceAkkaSerializer and "org.example.MySerializable" is the FQCN of your top-level serializable type (trait / abstract class) that is extended by Messages / Events / States which "org.example.ExampleSerializer" should serialize.
+
+Also, there are few optional configurations that you can set if you want to change the default behavior of Circe Akka Serializer:
+- `org.virtuslab.ash.circe.verbose-debug-logging` = `on` / `off`\
```

Comment:
Nit: you can probably just indent the next line instead of using `\`, this will be more readable in raw markdown (and the same in rendered HTML version)

Reply:
replaced `\` characters with blank lines - works fine